qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1692,3 +1706,107 @@ def __getattr__(self, name: str) -> Any:
         except KeyError:
             # pylint: disable=raise-missing-from
             raise AttributeError("Attribute %s is not defined" % name)
+
+    def _safe_serialize_jobs(self):
+        """Return serializable object for stored jobs"""
+        # Since Job objects are not serializable this removes
+        # them from the jobs dict and returns {job_id: None}
+        # that can be used to retrieve jobs from a service after loading
+        jobs = ThreadSafeOrderedDict()
+        for jid, _ in self._jobs.items():
+            jobs[jid] = None
+        return jobs
+
+    def _safe_serialize_figures(self):
+        """Return serializable object for stored figures"""
+        # Convert any MPL figures into SVG images before serializing
+        figures = ThreadSafeOrderedDict()
+        for name, figure in self._figures.items():
+            if isinstance(figure, pyplot.Figure):
+                figures[name] = plot_to_svg_bytes(figure)
+            else:
+                figures[name] = figure
+        return figures
+
+    def __json_encode__(self):
+        if any(not fut.done() for fut in self._job_futures.values()):
+            LOG.warning(
+                "Not all job futures have finished."
+                " Data from running futures will not be serialized."
+            )
+        if any(not fut.done() for fut in self._analysis_futures.values()):
+            LOG.warning(
+                "Not all analysis callbacks have finished."
+                " Results from running analysis will not be serialized."
+            )
+        json_value = {}
+        for att in [
+            "_metadata",
+            "_source",
+            "_service",
+            "_backend",
+            "_id",
+            "_parent_id",
+            "_type",
+            "_tags",
+            "_share_level",
+            "_notes",
+            "_data",
+            "_analysis_results",
+            "_analysis_callbacks",
```

Comment:
Do we really need to save these callbacks? Because these are not triggered until add method is explicitly called, thus having these data doesn't seem to give users a value though serialization of callback is really complicated (if we have rerun method probably it's worth having).

Reply:
These are the callback dataclasses that store the status and any error messages for finished callbacks, which are required for certain experiment data methods to work correctly.